### ISSUE TITLE
Discuss - 0037-disjoint-groups

### ISSUE THREAD

**Issue by afck** [maidsafe/rfcs#161]
[Rendered](https://github.com/maidsafe/rfcs/blob/master/text/0037-disjoint-groups/0037-disjoint-groups.md)

**Comment by dirvine** [comment]
In 
https://github.com/maidsafe/rfcs/blob/master/text/0037-disjoint-groups/0037-disjoint-groups.md#message-routing
I think we may need to consider that a node can (with DataChains) already proove it is a network member as well as group membership (via sending the link chain or checkpoint -> top of link chain, if the receiver knows the checkpoints). 
This would mean we would not need to accumulate at every step, but just push the message through the network and the receiver can confirm that message in it's entirety. Though as stated this is a future potential efficiency & security  improvement as it requires another RFC to be included int eh discussion. 
I just mention it here for completeness really.

**Comment by afck** [comment]
I don't understand how this could replace accumulation. I guess we won't put signatures of all messages we ever relayed as a group into the data chain?

I already assume in the algorithm that everyone in group `B` knows everyone in group `A` and believes them. But it's still necessary that a quorum in `A` signs the particular message that is being routed here, so that `B` knows that the message is genuine.

**Comment by dirvine** [comment]
Agreed, but the DataChain can handle Put Post Delete, so these can be sent across the network with no problem if backed by a signed message and at least part of the data chain. I think this is the biggest use case of messages that travel far (even safecoin creation). So in that case a single message would work and if all the rest of the group did the same or sent at least a note this was due, then we can bypass all routing accumulation for those.

**Comment by afck** [comment]
I don't see the difference between Put/Post/Delete and other messages for this purpose: In either case the challenge is to make sure that the recipient knows some checkpoint of the sender, and the sender includes the path from that checkpoint to the current group with the message.
Anyway, this is probably a discussion for another RFC. :-)

**Comment by dirvine** [comment]
Yes agreed, either checkpoint of whole linkchain and genesis etc. More related to DataChains itself though and i that then shows an efficiency gain there, all is good.

**Comment by afck** [comment]
I added some more details about the group merge flow: #164 
Please let me know whether you agree with these and whether they are detailed enough.

I expect that the flow might have to change again later, with data chains.

**Comment by Fraser999** [comment]
I agree and have merged this now.

**Comment by afck** [comment]
I added some more examples: https://github.com/maidsafe/rfcs/pull/166
